### PR TITLE
OSDP: Maintainers, samples and board updates

### PR DESCRIPTION
**Pull request by sidcha**
Ripping off individual commits that don't depend on anything from #46449.

### CODE REVIEW

File: boards/arm/nucleo_g474re/nucleo_g474re.dts
Comment (erwango) [inline boards/arm/nucleo_g474re/nucleo_g474re.dts:22]: Since this chosen symbol is quite specific and only used in a single board, I wonder if this should rather be added as an overlay in the samples.
I think this would benefit to the sample to illustrate how to use it.
Could also be set as an `app.overlay` file and be more generic using arduino node label.
```
	chosen {
		zephyr,osdp-uart = &arduino_serial;
         };
```

**Review by MaureenHelm — COMMENTED** [review]
Please address @erwango 's comments, but otherwise LGTM

**Comment by MaureenHelm** [comment]
cc: @adakus @r2r0

File: boards/arm/nucleo_g474re/nucleo_g474re.dts
Comment (sidcha) [reply boards/arm/nucleo_g474re/nucleo_g474re.dts:22]: @erwango, I like the idea of using `arduino_serial` node label. For boards that don't have it, is there a way to fallback to another?

For instance I use the [stm32_min_dev_(blue|black)](https://docs.zephyrproject.org/2.6.0/boards/arm/stm32_min_dev/doc/index.html) boards for testing but they don't have it.

**Review by erwango — CHANGES_REQUESTED** [review]
Please update `samples/subsys/mgmt/osdp/peripheral_device/sample.yaml` accordingly:
```
    depends_on: gpio arduino_serial
```

But this raises indeed and issue as in this case `stm32_min_dev_black` can't be used as integration platform.
Alternative is to drop the use of `arduino_serial`. This would limit the genericity of the sample, but adding a new overlay is not complex anyway.

File: samples/subsys/mgmt/osdp/control_panel/sample.yaml
Comment (erwango) [inline samples/subsys/mgmt/osdp/control_panel/sample.yaml:9]: filter is applied before dtc compilation, hence before the overlay is applied.
`dt_chosen_enabled("zephyr,osdp-uart")` will always return false.

File: samples/subsys/mgmt/osdp/peripheral_device/sample.yaml
Comment (str4t0m) [inline samples/subsys/mgmt/osdp/peripheral_device/sample.yaml:7]: a

File: samples/subsys/mgmt/osdp/control_panel/sample.yaml
Comment (str4t0m) [inline samples/subsys/mgmt/osdp/control_panel/sample.yaml:7]: This test now depends on `arduino_serial`, therefore can't use stm32_min_dev_black as an integration platform.
Please use a different board.

File: samples/subsys/mgmt/osdp/peripheral_device/sample.yaml
Comment (str4t0m) [inline samples/subsys/mgmt/osdp/peripheral_device/sample.yaml:7]: same here.

**Review by str4t0m — CHANGES_REQUESTED** [review]

File: boards/arm/nucleo_g474re/nucleo_g474re.dts
Comment (erwango) [reply boards/arm/nucleo_g474re/nucleo_g474re.dts]: For boards that don't have it, you can add a board specific overlay to the sample.

File: samples/subsys/mgmt/osdp/control_panel/sample.yaml
Comment (sidcha) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml:7]: Switched to `nucleo_g474re` at both places. Thanks!

File: samples/subsys/mgmt/osdp/control_panel/sample.yaml
Comment (sidcha) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml:9]: So how can I filter platforms that aren't supported? removing `dt_chosen_enabled("zephyr,osdp-uart")` widens the test space and so the test fails on unsupported platforms.

Comment (erwango) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml:9]: Well, in theory "filter" should list allow to filter out al platforms that don"'t provide required dependencies, so in theory it shouldn't.
But I agree that on this area, practice often defeats the theory.  Out of curiosity, do you have an example of unsupported platform that is not filtered out ?

Comment (sidcha) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml:9]: Just removing `dt_chosen_enabled("zephyr,osdp-uart")` from the filter causes test failure because the device tree lookup for "zephyr,osdp-uart".

I don't understand the failures as the app overlays should be applied at the time (after this filter is processed).

Comment (erwango) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml:9]: Can you provide an example of failing test command so I can have a try?

Comment (sidcha) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml:9]: Actually that is part of the problem; I cannot run the tests locally because something is broken in my python setup. Haven't spent the time fixing it.

Removing `dt_chosen_enabled("zephyr,osdp-uart")` caused CI builds to fail. I'll do that now.

Comment (sidcha) [reply samples/subsys/mgmt/osdp/control_panel/sample.yaml]: @erwango, can you help me with this CI failure please :)

**Comment by sidcha** [comment]
Maintainers and samples fix moved to separate PRs. The app overlay change needs a few more commits to clean up zephy-uart before it can be merged. So closing this PR.

**Pull request closed without merging**
